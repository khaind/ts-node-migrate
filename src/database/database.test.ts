import { MongoMemoryServer } from 'mongodb-memory-server';
import { createConnection } from '.';

let mongoServer: MongoMemoryServer;

describe('database implementation', () => {
  beforeAll(() => {
    mongoServer = new MongoMemoryServer();
  });

  afterAll(() => {
    mongoServer.stop();
  });

  test('should setup db conection successfully with correct url', async () => {
    const res = createConnection({
      url: await mongoServer.getUri(),
      dir: 'migrations',
      type: 'mongodb',
      tableName: 'migrations_changelog',
    });
    expect(res).toBeTruthy();
  });

  test('Invalid db type should throw error', async () => {
    const url = await mongoServer.getUri();
    expect(() =>
      createConnection({
        url,
        dir: 'migrations',
        type: 'invalidDb',
        tableName: 'migrations_changelog',
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
