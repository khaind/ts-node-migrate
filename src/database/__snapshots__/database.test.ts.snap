// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`database implementation Invalid db type should throw error 1`] = `undefined`;
